app/Database/Migrations/2020-07-12-143952_create_monsters.php

```diff
@@ -0,0 +1,95 @@
+<?php namespace App\Database\Migrations;
+
+use CodeIgniter\Database\Migration;
+
+class CreateMonsters extends Migration
+{
+	public function up()
+	{
+		/**
+		 * Monsters
+		 *
+		 * Notice the `dungeon_id` field, a reference back to our `dungeons` table. This
+		 * implies that each monster "belongs to" a single dungeon, especially since it
+		 * cannot be null.
+		 */
+		$fields = [
+			'name'       => ['type' => 'varchar', 'constraint' => 255],
+			'health'     => ['type' => 'int', 'null' => true],
+			'dungeon_id' => ['type' => 'int'],
+			'created_at' => ['type' => 'datetime', 'null' => true],
+			'updated_at' => ['type' => 'datetime', 'null' => true],
+			'deleted_at' => ['type' => 'datetime', 'null' => true],
+		];
+		
+		$this->forge->addField('id');
+		$this->forge->addField($fields);
+
+		$this->forge->addKey('name');
+		$this->forge->addKey('created_at');
+		
+		$this->forge->createTable('monsters');
+
+		/**
+		 * Abilities
+		 *
+		 * Abilities may be "cast" by anything that has the ability. Below we will make
+		 * a separate table to define who "has" an ability, since different monsters
+		 * can have the same ability (or heroes, for that matter).
+		 */
+		$fields = [
+			'name'       => ['type' => 'varchar', 'constraint' => 255],
+			'damage'     => ['type' => 'int', 'null' => true],
+			'cooldown'   => ['type' => 'int', 'null' => true],
+			'created_at' => ['type' => 'datetime', 'null' => true],
+			'updated_at' => ['type' => 'datetime', 'null' => true],
+		];
+		
+		$this->forge->addField('id');
+		$this->forge->addField($fields);
+
+		$this->forge->addKey('name');
+		$this->forge->addKey('created_at');
+		
+		$this->forge->createTable('abilities');
+		
+		/**
+		 * Abilities - Monsters
+		 *
+		 * Because we want monsters to have more than one ability, and many monsters to
+		 * have the same ability, we need a "join table" to define these "many to many"
+		 * relationships.
+		 */
+		$fields = [
+			'ability_id' => ['type' => 'int'],
```

Comment:
I don't recall off the top of my head, but when you use the `->addField('id')` that creates an `unsigned` int, right? If so, these keys should also be unsigned.

Reply:
You'll have to take that one up with Jim I'm afraid: https://github.com/codeigniter4/CodeIgniter4/pull/2278

😭

Reply:
Oh damn. Had forgotten about that. Ok, nvm for now :)